lib/metadataTypes/MetadataType.js

```diff
@@ -822,6 +825,9 @@ class MetadataType {
                 }
             } else if (singleRetrieve) {
                 // some types will return a single item intead of an array if the key is supported by their api
+                if (singleRetrieve.startsWith('id:')) {
```

Comment:
this naming convention of adding id: will not always apply. some use ObjectId, some use other id (querydefinitionId). Just wondering if another command would not be more appropriate

Reply:
the idea here is that you literally write `mcdev r cred/bu type id:asdadasdasd` to retrieve by id **asdadasdasd**. Meanwhile, the normal  `mcdev r cred/bu type dgfagfdsgdfgdf` would retrieve by key **dgfagfdsgdfgdf**.
Therefore, it's up to the user to add "id:" in front of an id to get this to work - if the type supports it. Not planning on adding that to any other type but journeys at this point. Therefore, this workaround seems better than adding a new command just for one type.

Reply:
i would rather you put flags if thats the case. using this makes it a naming convention not flag topic

Reply:
can you give an example? I don't understand

Reply:
ensure at least in docs that this is clear what it will actually work with.